hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/linux/resources/numa/NumaResourceAllocator.java

```diff
@@ -312,6 +312,8 @@ public synchronized void releaseNumaResource(ContainerId containerId) {
     for (NumaNodeResource numaNode : numaNodesList) {
       numaNode.releaseResources(containerId);
     }
+    // delete from NM State store
+    context.getNMStateStore().releaseAssignedResources(containerId, NUMA_RESOURCE_TYPE);
```

Comment:
Catch the IOException and throw ResourceHandlerException same as allocateNumaNodes method.

catch (IOException e) {
    throw new ResourceHandlerException(e);
}

And add throws ResourceHandlerException into releaseNumaResource() method.